packages/frontend/src/components/common/SpaceActionModal/CreateSpaceAddUser.tsx

```diff
@@ -0,0 +1,188 @@
+            usersSelected.map((userUuid) => ({
+                userUuid: userUuid,
+                firstName: '',
+                lastName: '',
+                role: null,
+            })),
+        );
+    }, [form, usersSelected]);
+    const renderUserShare: ItemRenderer<string> = useCallback(
+        (userUuid, { modifiers, handleClick, query }) => {
+            if (!modifiers.matchesPredicate) {
+                return null;
+            }
+            const user = organizationUsers?.find(
+                (userAccess) => userAccess.userUuid === userUuid,
+            );
+            if (!user) return null;
+
+            const isSelected = usersSelected.includes(userUuid);
+            const isYou = sessionUser?.userUuid === user.userUuid;
+            return (
+                <MenuItem2
+                    className={isSelected ? Classes.SELECTED : undefined}
+                    key={user.userUuid}
+                    disabled={isYou}
+                    text={
+                        <FlexWrapper key={user.userUuid}>
+                            <UserCircle>
+                                {getInitials(user.userUuid, organizationUsers)}
+                            </UserCircle>
+
+                            <PrimaryAndSecondaryTextWrapper $disabled={isYou}>
+                                {user.firstName || user.lastName ? (
+                                    <>
+                                        <PrimaryText $selected={isSelected}>
+                                            {user.firstName} {user.lastName}
+                                        </PrimaryText>
+
+                                        <SecondaryText>
+                                            {user.email}
+                                        </SecondaryText>
+                                    </>
+                                ) : (
+                                    <PrimaryText $selected={isSelected}>
+                                        {user.email}
+                                    </PrimaryText>
+                                )}
+                            </PrimaryAndSecondaryTextWrapper>
+                        </FlexWrapper>
+                    }
+                    onClick={(e) => {
+                        // Toggle user selected if selected
+                        if (usersSelected.includes(user.userUuid))
+                            setUsersSelected(
+                                usersSelected.filter(
+                                    (uuid) => uuid !== user.userUuid,
+                                ),
+                            );
+                        else handleClick(e);
+
+                        setSearchQuery('');
+                    }}
+                    shouldDismissPopover={false}
+                />
+            );
+        },
+        [usersSelected, organizationUsers],
+    );
+
+    return (
+        <>
+            <FlexWrapper>
+                <MultiSelect2
```

Comment:
can some of the components like this one be shared between CreateSpace modal and ShareSpace modal?

Reply:
ok cool 👍